llvm/include/llvm/IR/Attributes.h

```diff
@@ -775,6 +793,12 @@ class AttributeList {
   addAllocSizeParamAttr(LLVMContext &C, unsigned ArgNo, unsigned ElemSizeArg,
                         const std::optional<unsigned> &NumElemsArg);
 
+  /// Try to intersect this AttributeList with Other. Returns std::nullopt if
+  /// the two lists are inherently incompatible (imply different behavior, not
+  /// just analysis).
+  [[nodiscard]] std::optional<AttributeList>
+  intersectAttributes(LLVMContext &C, AttributeList Other) const;
```

Comment:
```suggestion
  intersectWith(LLVMContext &C, AttributeList Other) const;
```
For consistency with the AttributeSet method? Or is there a reason they have different names?

Reply:
Personally at least, prefer non-matching function names for avoid confusion, but ill change.